test/apps/iodemo/io_demo.cc

```diff
@@ -1366,16 +1372,17 @@ class DemoClient : public P2pDemoCommon {
         LOG << "terminate connection " << server_info.conn << " due to "
             << reason;
 
+        // Don't wait for any more completions on this connection
+        _num_sent -= get_num_uncompleted(server_info);
+
         // Remove connection pointer
         _server_index_lookup.erase(server_info.conn);
 
+        // Remove active servers entry
+        active_servers_remove(server_info.active_index);
```

Comment:
maybe better do `delete server_info.conn` first, and only then `_server_index_lookup.erase(server_info.conn);` ?
@hoopoepg can you pls post the full backtrace of the assertion?

Reply:
I can't: core files were deleted and no new runs on last week.
crash stack is looks like close_server->ucp_ep_close_nb + ucp_worker_progress->dispatch_io_message->assert